code/src/cljs/sixsq/nuvla/ui/edge_detail/views.cljs

```diff
@@ -1460,6 +1584,229 @@
+        [ui/Modal {:open       @show-modal?
+                   :close-icon true
+                   :on-close   close-fn}
+         [uix/ModalHeader {:header (@tr [:nuvlabox-add-playbook])
+                           :icon   "book"}]
+         [ui/ModalContent
+          [:<>
+           [ui/Form
+            [ui/FormInput {:label         (str/capitalize (@tr [:name]))
+                           :required      true
+                           :on-change     (ui-callback/input-callback
+                                            #(swap! form-data assoc :name %))}]
+            [ui/FormInput {:label         (str/capitalize (@tr [:description]))
+                           :required      false
+                           :on-change     (ui-callback/input-callback
+                                            #(swap! form-data assoc :description %))}]
+            [ui/FormSelect {:selection true
+                            :label  "Type"
+                            :fluid  false
+                            :default-value "MANAGEMENT"
+                            :options   available-types
+                            :on-change (ui-callback/value on-change-type)}]
+            [ui/FormInput {:label ["Enabled " (when @disabled-by-default?
+                                                (r/as-element
+                                                  [ui/Popup
+                                                   {:trigger        (r/as-element [ui/Icon {:name "info circle"}])
+                                                    :content        (@tr [:nuvlabox-playbook-emergency-info])}]))]}
+             [ui/Radio {:toggle    true
+                        :disabled  @disabled-by-default?
+                        :checked   (:enabled @form-data)
+                        :on-change #(do
+                                      (swap! form-data assoc :enabled (not (:enabled @form-data))))}]]
+            [ui/FormField {:label "Run"
+                           :required  true}]
+            "Shell script: "
+            [uix/EditorShell
+             ""
+             (fn [_editor _data value]
+               (ui-callback/input-callback
+                 (swap! form-data assoc :run value)))
+             true
+             "full-width"]]]]
+         [ui/ModalActions
+          [uix/Button
+           {:text     (@tr [:add])
+            :disabled (utils/form-add-playbook-incomplete? @form-data)
+            :primary  true
+            :on-click on-click-fn}]]]))))
+
+
+(defn TabPlaybooks
+  []
+  (let [tr             (subscribe [::i18n-subs/tr])
+        nuvlabox       (subscribe [::subs/nuvlabox])
+        playbooks      (subscribe [::subs/nuvlabox-playbooks])
+        selected-playbook (subscribe [::subs/nuvlabox-current-playbook])
+        run-changed?   (r/atom false)
+        run            (r/atom nil)
+        enabled-changed?  (r/atom false)]
+    (fn []
+      (let [n     (count @playbooks)
+            on-change-fn   (fn [k]
```

Comment:
This on-change-fn should be in Form 1 to avoid redefining it at each rendering

Reply:
done